File: app/src/main/java/de/herrmann_engel/rbv/activities/ManageFiles.java
package de.herrmann_engel.rbv.activities;

import android.os.Bundle;

import androidx.documentfile.provider.DocumentFile;
import androidx.recyclerview.widget.LinearLayoutManager;

import java.util.ArrayList;
import java.util.Objects;

import de.herrmann_engel.rbv.adapters.AdapterFilesManage;
import de.herrmann_engel.rbv.databinding.ActivityManageFilesBinding;
import de.herrmann_engel.rbv.db.utils.DB_Helper_Get;

public class ManageFiles extends FileTools {

    private ActivityManageFilesBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityManageFilesBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
    }

    @Override
    protected void onResume() {
        super.onResume();
        setRecView();
    }

    @Override
    protected void notifyFolderSet() {
        setRecView();
    }

    @Override
    protected void notifyMissingAction(int id) {

    }

    private void setRecView() {
        DocumentFile[] files = listFiles();
        ArrayList<DocumentFile> filesWithoutMedia = new ArrayList<>();
        DB_Helper_Get dbHelperGet = new DB_Helper_Get(this);
        for (DocumentFile file : files) {
            if (!dbHelperGet.existsMedia(file.getName()) && file.isFile() && !Objects.equals(file.getName(), ".nomedia")) {
                filesWithoutMedia.add(file);
            }
        }
        AdapterFilesManage adapter = new AdapterFilesManage(filesWithoutMedia);
        binding.recFilesManage.setAdapter(adapter);
        binding.recFilesManage.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: app/src/main/java/de/herrmann_engel/rbv/activities/ManageFiles.kt
package de.herrmann_engel.rbv.activities

import android.os.Bundle
import androidx.documentfile.provider.DocumentFile
import androidx.recyclerview.widget.LinearLayoutManager
import de.herrmann_engel.rbv.adapters.AdapterFilesManage
import de.herrmann_engel.rbv.databinding.ActivityManageFilesBinding
import de.herrmann_engel.rbv.db.utils.DB_Helper_Get

class ManageFiles : FileTools() {
    private lateinit var binding: ActivityManageFilesBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityManageFilesBinding.inflate(
            layoutInflater
        )
        setContentView(binding.root)
    }

    override fun onResume() {
        super.onResume()
        setRecView()
    }

    override fun notifyFolderSet() {
        setRecView()
    }

    override fun notifyMissingAction(id: Int) {}
    private fun setRecView() {
        val files = listFiles()
        if (files != null) {
            val filesWithoutMedia = ArrayList<DocumentFile>()
            val dbHelperGet = DB_Helper_Get(this)
            for (file in files) {
                if (!dbHelperGet.existsMedia(file.name) && file.isFile && file.name != ".nomedia") {
                    filesWithoutMedia.add(file)
                }
            }
            val adapter = AdapterFilesManage(filesWithoutMedia)
            binding.recFilesManage.adapter = adapter
            binding.recFilesManage.layoutManager = LinearLayoutManager(this)
        }
    }
}
